### PR TITLE
asking users to use their questions for research purposes

### PR DESCRIPTION
**Pull request by n4-t3**

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -32,6 +36,37 @@ export default function Page({ providers }: Props) {
...
+              isAdmin: user.isAdmin,
+              made: user.made.map(c => c.toString()),
+              madeOptions: user.madeOptions.map(c => c.toString()),
+              solved: user.solved.map(c => c.toString()),
+            })
+          )
+          if (user.studentID) {
+            dispatch(updateStudentID(user.studentID))
+          }
+          if (typeof user.allowDocumentation == 'boolean') {
```
Comment (jhw123) [inline src/pages/index.tsx:60]:
```suggestion
          if (user.allowDocumentation !== undefined) {
```

**Review by jhw123 — CHANGES_REQUESTED** [review]

File: src/pages/index.tsx [hunk 2]
```diff
@@ -32,6 +36,37 @@ export default function Page({ providers }: Props) {
     [setCode]
   )
 
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (n4-t3) [reply src/pages/index.tsx]: > I moved the logic to load user information from pages to `gnb.tsx`. Code at `gnb.tsx` should be changed

I think I made a mistake while merging updated it here https://github.com/kixlab/kuiz/pull/59/commits/de946c96cf4ad8f510b7f28397b28538db9c82b7

File: src/pages/index.tsx [hunk 3]
```diff
@@ -56,24 +91,41 @@ export default function Page({ providers }: Props) {
...
-          </InputSection>
-        </Sheet>
+        <>
+          <UpdateAllowDocumentationDialog modalState={askForDocumentation} submit={onUpdateAllowDocumentation} />
+          <Sheet>
+            <Header>Choose a Class or Enroll in a new Class</Header>
+            {classes.map(({ cid, name }, i) => (
+              <ClassButton key={i} onClick={onClassEnter(cid)}>
+                {name}
+                {/* ({code}) */}
```
[the remark this answers is not in the document]
Comment (n4-t3) [reply src/pages/index.tsx]: > I see some old code here. Perhaps due to incorrect conflict resolution?

I think I made a mistake while merging I have updated it here 
https://github.com/kixlab/kuiz/pull/59/commits/7e3962ffd3eae794e4d6a8fcd6a51f29728b4ecb

File: src/pages/index.tsx (hunk 1 above)
Comment (n4-t3) [reply src/pages/index.tsx]: updated here https://github.com/kixlab/kuiz/pull/59/commits/00a59d6ab95d68c19520d643d80548c4945d9053

File: src/pages/api/allowDocumentation.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { apiController } from '@utils/api'
+
+export interface AllowDocumentationParams {
```
[the remark this answers is not in the document]
Comment (n4-t3) [reply src/pages/api/allowDocumentation.ts]: > can we change the name of the API to `UpdateConsentState`? Or something more clear?

Yes, I have updated it here https://github.com/kixlab/kuiz/pull/59/commits/e6fd327b63dff80693b43ebe3630c61232432c3b :)

File: src/server/db/user.ts [hunk 5]
```diff
@@ -31,6 +32,10 @@ const UserSchema = new Schema<User>({
     type: String,
     required: false,
   },
+  allowDocumentation: {
+    type: Boolean,
+    required: false,
```
[the remark this answers is not in the document]
Comment (n4-t3) [reply src/server/db/user.ts]: > could you make this `required`?

yes updated it here https://github.com/kixlab/kuiz/pull/59/commits/530f26ba5742843f9e34a6093cc5764810cfb270 :)

File: src/components/Dialogs/updateAllowDocumentationDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import { FillButton } from '@components/basic/button/Fill'
+import { Modal } from './Modal'
+import { Label } from '@components/basic/Label'
+
+interface Props {
+  modalState: boolean
+  submit: (allow: boolean) => void
+}
+
+export const UpdateAllowDocumentationDialog = (props: Props) => {
```
[the remark this answers is not in the document]
Comment (n4-t3) [reply src/components/Dialogs/updateAllowDocumentationDialog.tsx:12]: > Could we use `window.confirm()` or embed checkbox on page instead? It seems like there are unnecessarily many Dialog components. We should reduce number of components (or abstract them into a single component) if possible for better maintenance

I have updated it here https://github.com/kixlab/kuiz/pull/59/commits/38bb9a6d82fd93032fd940f01f5983d1dcd058ea :)

File: src/server/db/user.ts [hunk 7]
```diff
@@ -4,6 +4,7 @@ import { Doc } from 'src/types/common'
 export interface User extends Doc {
   name: string
   studentID?: string
+  allowDocumentation?: boolean
```
Comment (jhw123) [inline src/server/db/user.ts:7]: one last thing! let's make the name consistent with the name in api `dataCollectionConsentState`

**Review by jhw123 — CHANGES_REQUESTED** [review]

File: src/server/db/user.ts (hunk 7 above)
Comment (n4-t3) [reply src/server/db/user.ts]: > one last thing! let's make the name consistent with the name in api `dataCollectionConsentState`

Ahh yes updated here https://github.com/kixlab/kuiz/pull/59/commits/32f0c91963cb3881b00da4d89e6b966d90d7b3f2 :)

**Review by jhw123 — APPROVED** [review]

**Pull request merged**